Library/Homebrew/software_spec.rb

```diff
@@ -391,23 +402,48 @@ def tab_attributes
 
   private
 
-  def github_packages_manifest_resource
-    return if @resource.download_strategy != CurlGitHubPackagesDownloadStrategy
+  # Resource for Docker manifests
+  def docker_manifest_resource(resource_name, basename, image_name, digest)
+    resource = Resource.new(resource_name)
+    resource.version(@version_rebuild)
+
+    resource.url("#{root_url}/#{image_name}/manifests/#{digest}", {
+      using: DockerRegistryDownloadStrategy,
+    })
+    resource.downloader.resolved_basename = basename
+    resource
+  end
+
+  # Resource for platform-specific manifest
+  def docker_bottle_manifest_resource(manifest)
+    return if @resource.download_strategy != DockerRegistryDownloadStrategy
+
+    @docker_bottle_manifest_resource ||= begin
+      @version_rebuild ||= GitHubPackages.version_rebuild(@resource.version, @rebuild)
```

Comment:
If this isn't GitHub Packages specific: the relevant class should probably be renamed?